pkg/flow/internal/controller/component.go

```diff
@@ -232,40 +232,40 @@ func (cn *ComponentNode) evaluate(scope *vm.Scope) error {
 	cn.doingEval.Store(true)
 	defer cn.doingEval.Store(false)
 
-	args := cn.reg.CloneArguments()
-	if err := cn.eval.Evaluate(scope, args); err != nil {
+	argsPointer := cn.reg.CloneArguments()
+	if err := cn.eval.Evaluate(scope, argsPointer); err != nil {
 		return fmt.Errorf("decoding River: %w", err)
 	}
 
 	// args is always a pointer to the args type, so we want to deference it since
 	// components expect a non-pointer.
-	argsCopy := reflect.ValueOf(args).Elem().Interface()
+	argsCopyValue := reflect.ValueOf(argsPointer).Elem().Interface()
 
 	if cn.managed == nil {
 		// We haven't built the managed component successfully yet.
-		managed, err := cn.reg.Build(cn.managedOpts, argsCopy)
+		managed, err := cn.reg.Build(cn.managedOpts, argsCopyValue)
 		if err != nil {
 			return fmt.Errorf("building component: %w", err)
 		}
 		cn.managed = managed
-		cn.args = argsCopy
+		cn.args = argsCopyValue
 
 		return nil
 	}
 
-	if reflect.DeepEqual(cn.args, args) {
+	if reflect.DeepEqual(cn.args, argsCopyValue) {
```

Comment:
Is this bug you mentioned? What do you mean about comparisons getting triggered?

Reply:
Yes, previously it was comparing a pointer to a value type.

Reply:
Sorry for the trouble, but can we check again? Looks like that both types were the same i.e. the component.Arguments interface.